src/main/java/core/mate/academy/service/ExcavatorProducerImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ExcavatorProducerImpl implements MachineProducer<Excavator> {
+    private List<Excavator> excavators;
```

Comment:
Don't forget about access modifiers for your custom fields in `Excavator`. The field `excavators` should be private to prevent direct access to it.